internal/plans/changes.go

```diff
@@ -641,12 +657,55 @@ func (c *Change) Encode(ty cty.Type) (*ChangeSrc, error) {
 		return nil, err
 	}
 
+	var beforeIdentityDV DynamicValue
+	var afterIdentityDV DynamicValue
+
+	identityTy := cty.DynamicPseudoType
+	if schema != nil {
+		identityTy = schema.Identity.ImpliedType()
+	}
+
+	// Identities can not have marks, this should be validated earlier
+	if !c.BeforeIdentity.IsNull() {
+		_, unsupportedBeforeIdentityMarks := c.BeforeIdentity.UnmarkDeepWithPaths()
```

Comment:
Identities are not configurable, and providers can't add marks, so there shouldn't be any way to get marks in here in the first place.

Reply:
Yeah I agree, I was trying to be extra careful, but it makes no sense here since there is no way for marks to exist in this context.